src/config.rs

```diff
@@ -56,13 +58,30 @@ fn merge_config(base: ConfigFile, override_config: ConfigEnv) -> Result<Config>
     })
 }
 
+fn get_config_file_path() -> Result<PathBuf> {
+    #[cfg(target_os = "macos")]
+    {
+        let home = std::env::var_os("HOME").ok_or(anyhow!("Unable to determine home directory"))?;
+        Ok(PathBuf::from(home)
+            .join(".config")
+            .join("adaptive-ml")
+            .join("adpt")
+            .join("config.toml"))
```

Comment:
the whole `.config` thing is part of the XDG spec which doesn't apply on macos, so I think it makes more sense just to have a `~/.adpt` and stick the config directly in there